CONTRIBUTING.md

```diff
@@ -0,0 +1,162 @@
+implementations, and it's also recommended that implementations quote the
+spec liberally and have automated testing that the quotes are correct, as
+[c-lightning
+does](https://github.com/ElementsProject/lightning/blob/master/tools/check-bolt.c).
+
+If your New Thing replaces the existing one, be sure to move the existing
+one to a Legacy subsection: new readers will want to go straight to the
+modern version.  Don't emulate the classic vsprintf man page:
+
+    Return Value
+   
+    ... return -1 if the output was truncated due to this limit.
+    (Thus until glibc 2.0.6. Since glibc 2.1 these functions follow the C99
+    standard and return...
+
+I deployed a bug once because I read the man page, but stopped before
+reading the aside which completely changed the answer.  It still makes
+me angry thinking about it!
+
+There's a [detailed style guide](.copy-edit-stylesheet-checklist.md) if you
+want to know how to format things, and we run a spellchecker in our [CI
+system](.travis.yml) as well so you may need to add lines to
+[.aspell.en.pws].
+
+### Writing The Requirements
+
+Some requirements are obvious, some are subtle.  They're designed to walk
+an implementer through the code they have to write, so write them as YOU
+develop YOUR implementation.  Stick with `MUST`/`SHOULD`/`MAY` and `NOT`:
+see [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt)
+
+Requirements are grouped into writer and reader, just as implementations
+are.  Make sure you define exactly what a writer must do, and exactly what
+a reader must do if it the writer doesn't do that!  A developer should
+never have to intuit reader requirements from writer ones.
+
+Note that the data doesn't have requirements: don't say `foo MUST be 0`,
+say `The writer MUST set foo to 0` and `The reader MUST fail the connection
+if foo is not 0`.
+
+Avoid the term `MUST check`: use `MUST fail the connection if` or `MUST
+fail the channel if` or `MUST send an error message if`.
+
+There's a subtle art here for future extensions: you might say `a writer
+MUST set foo to 0` and not mention it in the reader requirements, but it's
+better to say `a reader MUST ignore foo`.  A future version of the spec
+might define when a writer sets `foo` to `1` and we know that old readers
+will ignore it.
+
+`MAY` is a hint as to what something is for: an implementation may do
+anything not written in the spec anyway.  `MUST` is when not doing
+something will break the protocol or security.
+
+Requirements can be vague (eg. "in a timely manner"), but only as a last
+resort admission of defeat.  If you don't know, what hope has the poor
+implementer?
+
+### Creating Test Vectors
+
+For new low-level protocol constructions, test vectors are necessary.
+These have traditionally been lines within the spec itself, but the modern
+trend is to use JSON and separate files.  The intent is that they be
+machine-readable by implementations.
+
+For new inter-peer messages, a test framework is in development to simulate
```

Comment:
Could you maybe link to its repository (if there is one)?

Reply:
It'll be a PR RSN; it really needs to live in the main repository IMHO.